addons/gift_card/models/gift_card.py

```diff
@@ -13,10 +13,10 @@ class GiftCard(models.Model):
 
     @api.model
     def _generate_code(self):
-        return str(uuid4())[4:-8]
+        return '044' + str(uuid4())[4:-8][3:]
```

Comment:
why 044?

Reply:
[4:-8][3:] = [7:-8] ?

Reply:
It's because we want to use the barcode in the pos. So we need the barcode nomenclature at the start of the string.
I made a PR with few changes including what you mentioned:
 https://github.com/odoo/odoo/pull/75937